lesson_11/test_log_event.py

```diff
@@ -0,0 +1,197 @@
+"""Test cases for login event logging function."""
+
+
+import logging
+import os
+
+import pytest
+
+from homework10 import log_event
+
+
+class TestLogEvent:
+    """
+    A class for testing the log_event function.
+
+    This class is responsible for logging events in the login system.
+    All tests verify different login statuses
+    (successful, failed, expired, etc.) and log the results into a file.
+
+    Input data for the tests:
+    - username: 'test_user#'
+    - status: 'success', 'failed', 'expired', or an invalid status
+
+    Expected result:
+    - Logs should contain correct messages with appropriate logging levels
+    (info, warning, error).
+    """
+
+    # First 4 TCs are similar, all about status
+    # Test Case N1
+    def test_log_event_success(*args):
+        """Checking the correct logging of the successful login event."""
+        # Clear log-file if exists in folder. Use os.path to find file
+        if os.path.exists(log_file):
+            with open(log_file, 'w'):
+                pass
+
+        # Setup logger for the TC
+        _log = logging.getLogger('log_event')
+        _log.setLevel(logging.INFO)
+
+        # Clear previous handlers if exists
+        if _log.hasHandlers():
+            _log.handlers.clear()
+
+        # Create a new handler (write to file)
+        # Added levelname to see correct output
+        file_handler = logging.FileHandler(log_file)
+        file_handler.setFormatter(
+            logging.Formatter('%(asctime)s - %(levelname)s - %(message)s'),
+        )
+        _log.addHandler(file_handler)
+
+        # Call the function under test
+        log_event('test_user', 'success')
+
+        # Main logic here. Read log-file. Error handling added if no file
+        try:
+            with open(log_file, 'r') as lf:
+                log_txt = lf.read()
+        except FileNotFoundError:
+            raise AssertionError('Log file was not found.')
```

Comment:
please use pytest.fail() instead

Reply:
I can rewrite 'raise AssertionError' to pytest.fail(), but I'm not sure how to change all this logic to only pytest.fail(). Maybe put it in a separate function?